model/CHEM_DRV.F90

```diff
@@ -2097,22 +2111,29 @@ SUBROUTINE INIT_CHEM( grid )
        ENDIF
     ENDDO
     t_qlimit(:) = .true.
+
+    ! In the case of a cold restart, copy State_Chm into TrM as kg kg-1 for now
+    ! State_Met is not populated so we can't convert to kg
     TrM    = 0d0
     TrMom  = 0d0
-    
-    ! Copy State_Chm into TrM as kg kg-1 for now
-    ! State_Met is not populated so we can't convert to kg
-    DO N=1,NTM
-       DO L=1,LM
-          DO J=J_0,J_1
-             DO I=I_0,I_1
-                II = I - I_0 + 1
-                JJ = J - J_0 + 1
-                TrM( I, J, L, N ) = State_Chm%Species(N)%Conc(II,JJ,L)
-             ENDDO
-          ENDDO
-       ENDDO
-    ENDDO
+    IF (is_coldstart) THEN
+      DO N=1,NTM
+         DO L=1,LM
+            DO J=J_0,J_1
+               DO I=I_0,I_1
+                  II = I - I_0 + 1
+                  JJ = J - J_0 + 1
+                  TrM( I, J, L, N ) = State_Chm%Species(N)%Conc(II,JJ,L)
+               ENDDO
+            ENDDO
+         ENDDO
+      ENDDO
+    ELSE
+      ! In the case of a non-cold-restart, initialise GEOS-Chem from the Model E restart file
+      fid = par_open( grid, trim( 'fort.2.nc' ), 'read' )
+      CALL IO_CHEM(fid, 'read' )
+      call par_close( grid, fid )
```

Comment:
Does more logic need to come out of `Get_GC_Restart`.

We have derived types for simulation `State_Chm`, `State_Met`, `Input_Opt`, `State_Diag`. If I call any GEOS-Chem subroutine, it's going to be updating those derived types.

Need to check that `TrM` goes back into `State_Chm`.